test/unix/main.ml

```diff
@@ -853,6 +853,32 @@ module Throttle = struct
     ]
 end
 
+module Try_merge = struct
+  let test_during_async_merge () =
+    let log_size = 4 in
+    let* Context.{ rw; _ } = Context.with_empty_index ~log_size () in
+    let m = Semaphore.make false in
+    let hook =
+      Hook.v @@ function
+      | `Merge `Before ->
+          let not_previously_held = Semaphore.try_acquire m in
+          Alcotest.(check bool)
+            "Lock not previously held" true not_previously_held
+      | _ -> ()
+    in
+    for _ = 1 to log_size do
+      ignore (add_binding rw : binding)
+    done;
+    Log.app (fun m -> m "Triggering an implicit merge");
+    let _, merge_result = add_overflow_binding ~hook rw in
+    Log.app (fun m -> m "Running `try_merge` while a merge is ongoing");
+    Index.try_merge rw;
```

Comment:
shouldn't `try_merge` use the hook as well?

Reply:
Indeed, this mechanism for synchronising with the merge thread is broken. I'm now remembering why we didn't write tests this way to begin with... I'll propose something else as part of a different change.